TripKitUI/controller/TKSectionedAlertViewController.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  TKSectionedAlertViewController.swift
+//  TripKitUI-iOS
+//
+//  Created by Kuan Lun Huang on 15/3/18.
+//  Copyright © 2018 SkedGo. All rights reserved.
+//
+
+import UIKit
+import RxSwift
+import RxDataSources
+import SafariServices
+
+public class TKSectionedAlertViewController: UITableViewController {
+  
+  public var viewModel: TKSectionedAlertViewModel!
+  
+  private let disposeBag = DisposeBag()
+  
+  public static func newInstance() -> TKSectionedAlertViewController {
+    return TKSectionedAlertViewController(nibName: "TKSectionedAlertViewController", bundle: Bundle(for: self))
+  }
+        
+  override public func viewDidLoad() {
+    super.viewDidLoad()
+    
+    let nib = UINib(nibName: "TKRouteCell", bundle: Bundle(for: TKSectionedAlertViewController.self))
+    tableView.register(nib, forCellReuseIdentifier: "TKRouteCell")
+    tableView.rowHeight = UITableViewAutomaticDimension
+    tableView.estimatedRowHeight = 60
+    tableView.separatorStyle = .none
+    
+    bindViewModel()
+  }
+
+  override public func didReceiveMemoryWarning() {
+    super.didReceiveMemoryWarning()
+  }
+  
+  // MARK: -
+  
+  private func bindViewModel() {
+    guard viewModel != nil else { assert(false, "No view model found") }
+    
+    let dataSource = RxTableViewSectionedReloadDataSource<AlertSection>(configureCell: { (ds, tv, ip, item) -> UITableViewCell in
+      let cell = tv.dequeueReusableCell(withIdentifier: "TKRouteCell", for: ip) as! TKRouteCell
+      cell.route = item.alertGroup.route
+      cell.alertCount = item.alertGroup.mappings.count
+      cell.mode = item.mode      
+      return cell
+    })
+    
+    // table view section header
+    dataSource.titleForHeaderInSection = { ds, index in
+      return ds.sectionModels[index].header
+    }
+    
+    viewModel.sections
+      .observeOn(MainScheduler.instance)
+      .bind(to: tableView.rx.items(dataSource: dataSource))
+      .disposed(by: disposeBag)
+    
+    tableView.rx.itemSelected
+      .map { dataSource[$0] }
+      .subscribe(onNext: { [unowned self] in
+        self.didSelect($0)
+      })
+      .disposed(by: disposeBag)
```

Comment:
There's a neater way to do all of this, by passing the selection to the constructor of the view model.